imagebuildah/build.go

```diff
@@ -318,6 +318,34 @@ func buildDockerfilesOnce(ctx context.Context, store storage.Store, logger *logr
 		}
 		mainNode.Children = append(mainNode.Children, additionalNode.Children...)
 	}
+
+	// Check if any modifications done to labels
+	// add them to node-layer so it becomes regular
+	// layer.
+	// Reason: Docker adds label modification as
+	// last step which can be processed as regular
+	// steps and if no modification is done to layers
+	// its easier to re-use cached layers.
+	if len(options.Labels) > 0 {
+		for _, labelSpec := range options.Labels {
+			label := strings.SplitN(labelSpec, "=", 2)
+			labelLine := ""
+			key := label[0]
+			value := ""
+			if len(label) > 1 {
+				value = label[1]
+			}
+			labelLine = fmt.Sprintf("LABEL \"%s\"=\"%s\"\n", key, value)
```

Comment:
This doesn't look like it'll catch weirdness like values that include internal quotation marks of one kind or another.

Reply:
@nalind i tried this `buildah build --layers --label "label1=value'2" --from=alpine -t image1`  but this is failing  `buildah build --layers --label 'label1=value"2' --from=alpine -t image1` let me get a fix.

Reply:
@nalind I have changed this formatting with simple `%q` it autoescapes things for us tested both the cases, `"` will be autoescaped. 
```diff
-                       labelLine = fmt.Sprintf("LABEL \"%s\"=\"%s\"\n", key, value)
+                       labelLine = fmt.Sprintf("LABEL %q=%q\n", key, value)
```